modules/lib/src/main/resources/assets/js/app/inputtype/selector/ui/NewContentButtonUploaderHandler.ts

```diff
@@ -0,0 +1,127 @@
+import {ProgressBar} from '@enonic/lib-admin-ui/ui/ProgressBar';
+import {UploadItem} from '@enonic/lib-admin-ui/ui/uploader/UploadItem';
+import {Content} from '../../../content/Content';
+import {Element} from '@enonic/lib-admin-ui/dom/Element';
+import {FormItemEl} from '@enonic/lib-admin-ui/dom/FormItemEl';
+
+export class NewContentButtonUploaderHandler {
+
+    private static UPLOADING_CLASS: string = 'uploading';
+
+    private uploadItems: UploadItem<Content>[];
+
+    private readonly progressBar: ProgressBar;
+
+    private readonly parentElement: Element;
+
+    private readonly progressMap: Map<string, number> = new Map<string, number>();
+
+    private currentlyUploadedPercentage: number = 0; // sum of uploaded percents of ALL items to compare with absolute value
+
+    private itemsUploadedOrFailed: number = 0;
+
+    private itemUploadedHandler?: (content: Content) => void;
+
+    private itemUploadFailedHandler?: (item: UploadItem<Content>) => void;
+
+    constructor(parentElement: Element) {
+        this.progressBar = new ProgressBar(0);
+        this.parentElement = parentElement;
+
+        this.parentElement.appendChild(this.progressBar);
```

Comment:
Maybe this should be done in `handleItemProgress` and then removed from `this.parentElement` once the progress is either finished or failed? We don't need to keep it in the DOM unless it's visible. Then we won't need `UPLOADING_CLASS` and CSS tweaks either.

Reply:
We need CSS class at least to hide other elements contained in a parent element